src/eda_server/api/websocket.py

```diff
@@ -220,14 +223,64 @@ async def handle_workers(websocket: WebSocket, data: dict, db: AsyncSession):
         )
 
 
+@cached(cache=Cache.MEMORY)
+async def get_job_instance_id(uuid: str, db: AsyncSession):
+    query = select(models.job_instances).where(
+        models.job_instances.c.uuid == uuid
+    )
+    job_instance = (await db.execute(query)).first()
+    return job_instance.id
+
+
+async def insert_job_instance_events(
+    bulk_job_instance_events: asyncio.Queue, db_session_factory
+):
+    async with db_session_factory() as db:
+        query = insert(models.job_instance_events).values(
+            [
+                await bulk_job_instance_events.get()
+                for i in range(bulk_job_instance_events.qsize())
+            ]
+        )
+        await db.execute(query)
+        await db.commit()
+
+
+async def insert_job_instance_hosts(
+    bulk_job_instance_hosts: asyncio.Queue, db_session_factory
+):
+    async with db_session_factory() as db:
+        query = insert(models.job_instance_hosts).values(
+            [
+                await bulk_job_instance_hosts.get()
+                for i in range(bulk_job_instance_hosts.qsize())
```

Comment:
Shouldn't this also reference the `queue` attribute?
```suggestion
                await bulk_job_instance_hosts.queue.get()
                for i in range(bulk_job_instance_hosts.queue.qsize())
```